mock_ros2_project/service_server.py

```
@@ -0,0 +1,57 @@
+#!/usr/bin/env python3
+"""
+ROS2 Service Server Example
+
+This node demonstrates how to create a service server that provides
+an addition service using the AddTwoInts service type.
+"""
+
+import rclpy
+from rclpy.node import Node
+from example_interfaces.srv import AddTwoInts
+
+
+class ServiceServer(Node):
+    """Service server node that provides addition service."""
+    
+    def __init__(self):
+        super().__init__('service_server')
+        
+        # Create service
+        self.srv = self.create_service(
+            AddTwoInts,
+            'add_two_ints',
+            self.add_two_ints_callback
+        )
+        
+        self.get_logger().info('Service server ready to add two integers.')
+
+    def add_two_ints_callback(self, request, response):
+        """Service callback that adds two integers."""
+        response.sum = request.a + request.b
+        
+        self.get_logger().info(
+            f'Incoming request: a={request.a}, b={request.b}, '
+            f'sending back response: sum={response.sum}'
+        )
+        
+        return response
+
+
+def main(args=None):
+    """Main function to run the service server."""
+    rclpy.init(args=args)
+    
+    service_server = ServiceServer()
+    
+    try:
+        rclpy.spin(service_server)
+    except KeyboardInterrupt:
+        pass
```

Silently swallowing KeyboardInterrupt hides shutdown reason; log a friendly shutdown message instead.